lumbermill-aws/src/main/java/lumbermill/internal/aws/S3ClientImpl.java

```diff
@@ -113,13 +121,9 @@ public void init() {
                         LOGGER.debug("Deleted s3 file s3://{}/{} successfully", sBucket, sKey);
                     }
                 })
-                .doOnUnsubscribe(() -> {
+                .doAfterTerminate(() -> {
```

Comment:
I'm not so sure this is the correct thing to do.  We only want to delete the downloaded (local, temporary) file at the end of the pipeline, but the terminal event can come before that.

So perhaps doOnUnsubscribe is still the correct event on which to remove the downloaded file?